homeassistant/components/garmin_connect/__init__.py

```diff
@@ -86,23 +86,66 @@ def __init__(self, hass, client):
         """Initialize."""
         self.client = client
         self.data = None
+        self._activity_data = None
+        self._body_composition_data = None
 
     @Throttle(MIN_SCAN_INTERVAL)
     async def async_update(self):
         """Update data via library."""
         today = date.today()
 
         try:
-            self.data = self.client.get_stats(today.isoformat())
+            self._activity_data = self.client.get_stats(today.isoformat())
         except (
             GarminConnectAuthenticationError,
             GarminConnectTooManyRequestsError,
         ) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get activity data: %s", err
+            )
             return
         except (GarminConnectConnectionError) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get activity data: %s", err
+            )
             return
         except Exception:  # pylint: disable=broad-except
-            _LOGGER.error("Unknown error occurred during Garmin Connect get stats")
+            _LOGGER.error(
+                "Unknown error occurred during Garmin Connect get activity data"
+            )
             return
+
+        try:
+            self._body_composition_data = self.client.get_body_composition(
+                today.isoformat()
+            )["totalAverage"]
+        except (
+            GarminConnectAuthenticationError,
+            GarminConnectTooManyRequestsError,
+        ) as err:
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get body composition data: %s",
+                err,
+            )
+            return
+        except (GarminConnectConnectionError) as err:
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get body composition data: %s",
+                err,
+            )
+            return
+        except KeyError:
+            _LOGGER.error(
+                "Keyerror occurred during Garmin Connect get body composition data"
```

Comment:
Please make it more explicit what the reason for the error is, eg by including the error argument in the log.